src/sentaku/context.py

```diff
@@ -186,4 +186,4 @@ def __get__(self, instance, owner):
 
         bound_method = implementation.__get__(instance, type(instance))
         with ctx.use(choice, frozen=True):
-            return bound_method()
```

Comment:
if this change works, then we do the wrong thing and the obtaining of the value needs to happen within the with block
else the actual action happens in the outer context instead of the inner context thus resulting in wrong actions getting taken